tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -252,26 +263,28 @@ TEST(SGSparseMatrix, get_transposed_more_vectors)
 
 TEST(SGSparseMatrix, from_dense)
 {
-    //GenerateSparseMatrix cannot templated to generate dense matrix,
-    //since () operator are defined differently for sparse and dense matrix
-    CRandom randGenerator(0);
-    float64_t sparseLevel=0.1;
 
+    const float64_t sparseLevel=0.1;
     const index_t numberOfFeatures=50;
     const index_t numberOfVectors=100;
-    SGMatrix<float64_t> denseMatrix(numberOfFeatures, numberOfVectors);
-    for (index_t featIndex=0; featIndex<numberOfFeatures; ++featIndex)
-        for (index_t vecIndex=0; vecIndex<numberOfVectors; ++vecIndex)
-        {
-            float64_t randomNumber=randGenerator.random(0.0,1.0);
-            if (randomNumber<=sparseLevel)
-                denseMatrix(featIndex,vecIndex)=randomNumber;
-        }
+    const index_t randSeed=0;
+
+    SGMatrix<float64_t> denseMatrix(numberOfVectors, numberOfFeatures); denseMatrix.zero();
+    GenerateMatrixAccrodingToSparseConvention<SGMatrix<float64_t> >(sparseLevel,
+                                                                    numberOfFeatures,
+                                                                    numberOfVectors,
+                                                                    randSeed,
+                                                                    &denseMatrix);
```

Comment:
Why is GitHub marking this in another colour? Something different with the indentation (using spaces instead of tabs) maybe?